packages/web-components/src/menu/menu.ts

```diff
@@ -0,0 +1,64 @@
+import { FASTMenu } from '@microsoft/fast-foundation';
+import { attr } from '@microsoft/fast-element';
+
+/**
+ * The base class used for constructing a fluent-menu custom element
+ * @public
+ */
+export class Menu extends FASTMenu {
+  /**
+   * sets menuitem styles to align content when checkmarks are present
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: checkmarks
+   */
+  @attr({ mode: 'boolean' })
+  public checkmarks: boolean = false;
+  protected checkmarksChanged(): void {
+    if (!this.$fastController.isConnected) {
+      return;
+    }
+    this.menuItems?.filter(this.isMenuItemElement).forEach((item: HTMLElement, index: number) => {
+      if (this.checkmarks) {
+        item.setAttribute('checkmark', '');
+      } else {
+        item.removeAttribute('checkmark');
+      }
+    });
+  }
```

Comment:
I'd like to avoid doing something like this if we can. If we need to do something like this, you're setting it based on the role. As a user of the components, it would be incredibly redundant for me to have to both enumerate the role of the element *and* then set a flag on the parent.

Reply:
I agree that it's redundant but not sure how else to approach this. This is the same pattern `react` is following and provides a way to target non radio/checkbox `MenuItems` that appear in Menu that does also contain a subset of `menuitemradio` / `menuitemcheckbox` so that all the content in the `Menu` can be visually aligned. It's providing the same functionality as the `icons` attribute which aligns content when only a subset of `MenuItems` contain icons.

Reply:
I'm not sure I follow - are we applying a redundancy here to mimic react? The current implementation has an affordance to meet this design criteria. What I'm suggesting is that if you need something to hook to *in the menu item*, because that is where the styling is being applied, you have a _role_ attribute which is reflected to the :host. If that is not working, we can look to have someone pair and look at the implementation. But I don't think there is a scenario where you would not show these visually, so at the very least, we should not expose an attribute. If we need to use _both_ `icons` and some other thing, I'd recommend that we take a singular approach that handles both naturally rather than expecting developers to pass icon because they slotted an icon, checkmarks while enumerating a role of checkbox or radio, etc.

Reply:
I'm not quite following..

The problem I'm trying to solve is that a MenuItem needs to be conditionally styled based on the context of the other MenuItems it appears with in a Menu. 

So in each of these examples Item 3 is going to be styled differently based on the roles or inclusion of icons in other menu items in the same menu.


```html
<Menu>
     <MenuItem> Item 1 </MenuItem>
     <MenuItem> Item 2 </MenuItem>
     <MenuItem> Item 3 </MenuItem>
 </Menu>
 ```
```html
<Menu>
     <MenuItem role="radio"> Radio 1 </MenuItem>
     <MenuItem role="radio"> Radio 2 </MenuItem>
     <MenuItem> Item 3 </MenuItem>
 </Menu>
 ```
```html
<Menu>
     <MenuItem role="radio"></Icon> Radio 1 </MenuItem>
     <MenuItem role="radio"></Icon> Radio 2 </MenuItem>
     <MenuItem> Item 3 </MenuItem>
 </Menu>
 ```